thriftclient/client.go

```diff
@@ -29,13 +22,11 @@ type MonitoredClient struct {
 
 // NewMonitoredClient returns a pointer to a new MonitoredClient that wraps the
 // given thrift.TClient calls in thrift client spans.
-//
-// If c is nil, will return a nil client and an error rather than panicking.
-func NewMonitoredClient(c thrift.TClient) (*MonitoredClient, error) {
+func NewMonitoredClient(c thrift.TClient) *MonitoredClient {
 	if c == nil {
-		return nil, ErrNilTClient
+		panic("NewMonitoredClient received a nil client")
```

Comment:
Oh I thought you meant let it panic naturally :badpokerface:

Reply:
😅 

I prefer to let it panic sooner if it's something that will panic if you ever actually use it, I'm fine to remove this too.

Reply:
I would lean against calling `panic` explicitly from the library code, but I don't feel strongly about it.

Reply:
I also don't feel that strongly about it, I'll remove it.